### DISCUSSION TITLE
TTGO LoRa32 -OLED v2.1.6; Clone remote controller; Send and receive raw data

### DISCUSSION THREAD

**Discussion by forest-roby** [jgromes/RadioLib#275] [Q&A]
Dear All,

I try to copy/clone a remote controller in order to integrated to a home automation system. I did this kind of thing before with a 433Mhz remote controller - it was not easy (rc-switch didnt work) but at the end I succeded.
Now it is a different remote:
- 3 buttons(up, stop, down)
- 868.9Mhz
- Recorded the signal with Universal radio hacker (I got the idea from here, thankfully I have an sdr device)
- I can make a picture of the signal with SDR# or SDRconsole
- Remote IC: EFR32 FG1P133GG (dual band)
- frequency deviation and bitrate is unknown for now

With the 433Mhz remote: I had at home 433Mhz transiver and receiver modules. I was able to receive the RAW data from the remote and I could duplacatet/copy it. Those modules have three pins: VCC, GND and DATA - not too compicated.

I have no experience with 868Mhz at all and I bought this decive: TTGO LoRa32 -OLED v2.1.6, hoping that I'll be abel to do the some with this than previosly with 433Mhz modules. So far, it is not working at all. 
I couldn't receive any signal with this device from the remote, and I couldn't send any square wave signal out from this device. total disaster.

My best result so far is with Universal radio hacker which is a start but...

Can you confirm my suspission that this: TTGO LoRa32 -OLED v2.1.6 device is not suitable for duplacating the remote's signals?
Is your library supporting this device?
What kind of 868Mhz (arduino compatible) module can I buy that does the job?

Any help would be appreciated.
Thank you
Best regards,

Robert

**Comment by jgromes** [comment]
> I couldn't send any square wave signal out from this device.

No wonder, TTGO LoRa32 works completely differently to the "raw" 433 MHz transmitters you probably used before. All the wireless modules in this library - including the one on TTGO - were designed to transmit packets.

> Can you confirm my suspission that this: TTGO LoRa32 -OLED v2.1.6 device is not suitable for duplacating the remote's signals?

No idea - depends on the signal the remote is sending. If I recall correctly, TTGO LoRa32 uses SX1278 as the LoRa module, so it is supported by RadioLib. Assuming the remote is transmitting something common like FSK or OOK, you should be able to receive it, in theory. In practice, there might be a lot of hoops to jump through, such as encoding and encryption. SDR is your friend here, but you willl have to do a lot of reverse engineering to figure out the bit rate, frequency deviation, sync word, packet format, encoding, encryption, ...

TL;DR: It's probably not worth the effort. But it's hard to say since you didn't post much information about the remote. Googling the IC part number you provided didn't yield much info.
